csrc/python_frontend/python_bindings.cpp

```diff
@@ -1838,6 +1859,48 @@ void initNvFuserPythonBindings(PyObject* module) {
       py::arg("pad_widths"),
       py::arg("value") = py::none(),
       py::return_value_policy::reference);
+  nvf_ops.def(
+      "take_along_axis",
+      [](FusionDefinition::Operators& self,
+         Tensor arg1,
+         Tensor index,
+         int64_t dim) -> Tensor {
+        FUSER_PERF_SCOPE("Operators.take_along_axis");
+        TORCH_CHECK(
+            self.validUse(), "Attempting to add to a completed definition!");
+        FusionDefinition* fd = self.fusion_definition;
+        Tensor output = fd->defineTensor(arg1.dims);
+        fd->defineRecord(new TakeAlongAxisOpRecord(
+            {
+                fd->recordingState(arg1()),
+                fd->recordingState(index()),
+            },
+            {fd->recordingState(output())},
+            dim));
+        return output;
+      },
+      R"pbdoc(
+        Index arg1 in dim at positions given by index.
+
+        This operation is very similar to :meth:'gather' but enforces that all
+        dimensions other than dim must be equal between arg1 and index.
+
+        Args:
+            arg1 (Tensor): Tensor of shape `(Ni...,M,Nk...)` where `M` is the
+                extent of `arg1` in the dimension `dim`.
+            index (Tensor): Tensor of dtype `DataType::Int` of shape
+                `(Ni...,J,Nk...)`.
+            dim (int): Which position to index along.
+
+        Returns:
+            (Tensor): Tensor of same dtype as `arg1` and of shape
+                `(Ni...,J,Nk...)` where the element at position `(i...,j,k...)`
+                is equal to `arg1[i,...,index[i,...,j,k,...],k,...]`.
+      )pbdoc",
```

Comment:
@kevinstephano I know we don't yet have any docstrings, but I added a couple here to distinguish from `gather`. It maybe should use different formatting for the indices; I was following numpy's docs https://numpy.org/doc/stable/reference/generated/numpy.take_along_axis.html.

Reply:
While we're on this topic, I guess at some point we should consider setting up sphinx and adding nvfuser on readthedocs.org.